### PR TITLE
classPrefix option

### PR DESCRIPTION
**Pull request by bigdot**
- added `classPrefix` option for applications and frameworks not yet namespace ready
- added 'baseClass' option for all classes to extend
- fixed methods having a null return type

### CODE REVIEW

**Comment by bigdot** [comment]
@kasperg can you check this out ?

**Comment by kasperg** [comment]
@bigdot Thanks for the suggestion and apologies for the late reply.

I will not accept this pull request in its current form. PHP has had namespace support for over 7 years now. I do not want to introduce complexity in this project to handle legacy applications or frameworks.

Regarding base classes I think traits is a better way to achieve the same thing. See #257 for discussion and #260 for a PR with an implementation of this.

You also mention "fixed methods having a null return type". It is unclear to me what part of the PR handles this. If it has merrit on its own please submit a separate pull request for this.

File: src/Service.php [hunk 1]
```diff
@@ -65,7 +71,13 @@ public function __construct(ConfigInterface $config, $identifier, array $types,
         $this->operations = array();
         $this->types = array();
         foreach ($types as $type) {
-            $this->types[$type->getIdentifier()] = $type;
+            if ($type !== null) {
+                $this->types[$type->getIdentifier()] = $type;
+            }
```
Comment (bigdot) [inline src/Service.php]: @kasperg For WSDLs that have nil/void return types, it will issue a fatal error without this `if`.

**Comment by bigdot** [comment]
@kasperg Hey.

I don't see this PR as added complexity. It's just another option that was added. The features to make that happen were already there in the generator. There are just a couple of params added, nothing more.

There are still a couple of php products widely used, that don't use namespaces. 
I made this patch for a Magento 1.9 project, and by the looks of it, magento 2 (which uses namespaces) is still 2 years from wide use.

Added a comment with the null return type thingie.
Let me know if you want me to decouple it in another PR.

File: src/Service.php (hunk 1 above)
Comment (kasperg) [inline src/Service.php]: This was fixed in #301.

**Comment by kasperg** [comment]
I do not know Magento so bear with me.

As far as I can tell Magento 1.9 supports PHP >=5.4 which supports namespacing. What is preventing you from requiring the generated autoloader (assuming Magento 1.x does not use one) and using generated classes *with* a namespace?

Versio 2.x of this library used to support class prefixes (and suffixes). Please see [the update regarding 3.x](https://github.com/wsdl2phpgenerator/wsdl2phpgenerator/blob/master/docs/whats-new-in-3.0.md#obsolete-configuration-options-removed) and the associated issues/PRs about why this option was removed.

**Comment by mikaelcom** [comment]
@bigdot  if you want to prefix your generated classes, use another project such as the [PackageGenerator](https://github.com/WsdlToPhp/PackageGenerator) project which offers prefix, suffix, namespacing and many other options ;)

File: src/Service.php (hunk 1 above)
Comment (bigdot) [inline src/Service.php]: cool

**Comment by bigdot** [comment]
Heya,

The php version supports namespaces indeed, and that is one solution, to use namespaced generated classes. However, the factories in magento 1.9 do not support namespaces, eg: `Mage::getModel('mymodule/my_wsdl_type_Customer')`. You can also register an autoloader for your types but then you add a little complexity in the module.

I, for one, am in favor of removing obsolete options, and code refactored and adapted for the current technology. I think this magento 1.9 is maybe the only one left (still in wide use) that would still justify backward compatibility, and as i see it, it will be obsolete by the end of next year.

I personally have no situation with the project since i'm versioning the generated classes (this might change in time though, but it's no biggie). 

The patch in here helped me and i thought it could help others in my situation. You can leave it out.

**Pull request closed without merging**